include/lsst/afw/geom/Transform.h

```diff
@@ -189,6 +189,28 @@ class Transform {
      */
     Eigen::MatrixXd getJacobian(FromPoint const &x) const;
 
+    /**
+     * Concatenate two Transforms.
+     *
+     * @tparam LastEndpoint the starting Endpoint of the Transform that feeds
+     *                      into this one
+     * @param before the Transform to apply before this one
+     * @returns a Transform that first applies `before` to its input, and then
+     *          this Transform to the result. Its inverse shall first apply the
+     *          inverse of this Transform, then the inverse of `before`.
+     *
+     * @throws InvalidParameterErrror Thrown if `this->getFromEndpoint()` does
+     *                                not match `before.getToEndpoint()`.
```

Comment:
I think "number of axes" is a bit too specific. Can we be sure that's the only way that two Endpoints of the same type will differ?